### PR TITLE
Docs separate pages

### PR DESCRIPTION
**Pull request by AydinHassan**
Docs are on separate pages with some some classes to represent the docs with back, forward stuff automatically rendered, there's loads of stuff I don't really like about this yet.

### CODE REVIEW

**Comment by AydinHassan** [comment]
@mikeymike have a check over this, don't merge yet tho, might still change some stuff.

**Comment by mikeymike** [comment]
hey man sorry it's took so long, hope you've had some good time off anyway ! 😎 

This looks pretty fucking sweet to me dude, I know how much of a perfectionist you are but the only thing I'm really picking up as something I'm not that keen on is the mutability of the sections/groups etc

I think we could probably make these immutable and have the logic for getting the next / previous and home links from either the documentation class or a separate class. 

I'm thinking down the lines of... 

```
function getPreviousSection(DocumentationSection $section) : DocumentationSection {
      $offset = array_search($section->getName(), array_keys($sections)) - 1;
      
      if ($offset < 0) {
          // handle no previous
      }
      
      return array_slice($sections, $offset, 1);
}
```

Of course next would be pretty trivial too and home would just be returning the parent group I guess. 

what do you think?

**Comment by AydinHassan** [comment]
hey @mikeymike I've updated based on your suggestions. I think it's a bit nicer now. I still feel like the whole thing is a bit too complicated - what do you think?

**Comment by AydinHassan** [comment]
@mikeymike cool - think I'm happy with this now - what do you think?

File: src/Documentation.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    private $groups = [];
+
+    /**
+     * @param DocumentationGroup $group
+     */
+    public function addGroup(DocumentationGroup $group)
+    {
+        $this->groups[] = $group;
+    }
+
```
Comment (mikeymike) [inline src/Documentation.php]: jw... are we bothered about docblocks in 7 ?

Comment (AydinHassan) [inline src/Documentation.php]: Nah I don't really think it's worth it, just duplication - what do you think?

Comment (mikeymike) [inline src/Documentation.php]: that's what I was thinking, 7 is self documenting (when you use it like that anyway)

only thing I'd say is  consistency... eg method above has it and others don't .. happy to go with either tbh

Comment (AydinHassan) [inline src/Documentation.php]: sure - i'll remove it

**Comment by mikeymike** [comment]
@AydinHassan looks good to me bro 👍 ... In terms of complication I think it's fine I can't see a way of making it simpler without stripping out features 😄

**Comment by AydinHassan** [comment]
Sweet, I simplified it a fair bit now anyway, cheers for suggestions 😀

**Pull request merged**